### PR TITLE
Move cancellation hook to utils.js in preparation for inactive session code [SATURN-1243]

### PR DESCRIPTION
**Pull request by petesantos**
Moving this hook into utils.js in preparation for adding a polling hook for the inactive timeout.

The inactive timeout hook will utilize useCancellation. If the polling hook is added to Utils this will create a circular dependency between the ajax an utils modules, this change will prevent that.

Tested by:
1. Clicking through pages on the site
2. Throttling the speed down to "Slow 3g" in the dev tools network tab and
  a. closing modals before ajax requests completed and observing the cancellation in the network tools
  b. Navigating away before an ajax request was completed and observing the cancellation in the network tools

### CODE REVIEW

File: src/pages/Clusters.js [hunk 1]
```diff
@@ -8,14 +8,14 @@ import { icon } from 'src/components/icons'
...
-import { Ajax, useCancellation } from 'src/libs/ajax'
+import { Ajax } from 'src/libs/ajax'
 import { getUser } from 'src/libs/auth'
 import { clusterCost, currentCluster } from 'src/libs/cluster-utils'
 import colors from 'src/libs/colors'
 import { withErrorIgnoring, withErrorReporting } from 'src/libs/error'
 import * as Nav from 'src/libs/nav'
 import * as Style from 'src/libs/style'
-import { delay, formatUSD, makeCompleteDate, useGetter, useOnMount, withBusyState } from 'src/libs/utils'
+import { delay, formatUSD, makeCompleteDate, useCancellation, useGetter, useOnMount, withBusyState } from 'src/libs/utils'
```
Comment (zarsky-broad) [inline src/pages/Clusters.js:18]: This should become `* as Utils`, we could do it later or we could do it now.

File: src/pages/library/Code.js [hunk 2]
```diff
@@ -6,14 +6,14 @@ import { centeredSpinner } from 'src/components/icons'
...
-import { Ajax, useCancellation } from 'src/libs/ajax'
+import { Ajax } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
 import { getConfig } from 'src/libs/config'
 import { withErrorReporting } from 'src/libs/error'
 import { getAppName, returnParam } from 'src/libs/logos'
 import * as StateHistory from 'src/libs/state-history'
 import * as Style from 'src/libs/style'
-import { useOnMount, withBusyState } from 'src/libs/utils'
+import { useCancellation, useOnMount, withBusyState } from 'src/libs/utils'
```
Comment (zarsky-broad) [inline src/pages/library/Code.js:16]: This should become `* as Utils`, we could do it later or we could do it now.

File: src/pages/workspaces/workspace/RequestAccessModal.js [hunk 3]
```diff
@@ -4,9 +4,9 @@ import { div, h, span, table, tbody, td, th, thead, tr } from 'react-hyperscript
 import { ButtonPrimary, Link } from 'src/components/common'
 import { centeredSpinner, icon } from 'src/components/icons'
 import Modal from 'src/components/Modal'
-import { Ajax, useCancellation } from 'src/libs/ajax'
+import { Ajax } from 'src/libs/ajax'
 import { withErrorReporting } from 'src/libs/error'
-import { cond, useOnMount, withBusyState } from 'src/libs/utils'
+import { cond, useCancellation, useOnMount, withBusyState } from 'src/libs/utils'
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/RequestAccessModal.js:9]: This should become `* as Utils`, we could do it later or we could do it now.

File: src/pages/workspaces/workspace/SubmissionQueueStatus.js [hunk 4]
```diff
@@ -3,9 +3,9 @@ import _ from 'lodash/fp'
 import { Fragment, useState } from 'react'
 import { h, table, tbody, td, tr } from 'react-hyperscript-helpers'
 import { spinner } from 'src/components/icons'
-import { Ajax, useCancellation } from 'src/libs/ajax'
+import { Ajax } from 'src/libs/ajax'
 import { withErrorReporting } from 'src/libs/error'
-import { useOnMount, withBusyState } from 'src/libs/utils'
+import { useCancellation, useOnMount, withBusyState } from 'src/libs/utils'
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/SubmissionQueueStatus.js:8]: This should become `* as Utils`, we could do it later or we could do it now.

**Review by zarsky-broad — APPROVED** [review]

File: src/pages/Clusters.js (hunk 1 above)
Comment (petesantos) [reply src/pages/Clusters.js:18]: I'll take care of that in a separate PR - thanks!

**Pull request merged**